FireBrick/UI/AuthViewController/AuthViewController.swift

```diff
@@ -56,102 +53,110 @@ class AuthViewController: UIViewController, GIDSignInUIDelegate {
+    @objc
+    private func emailButtonTap(_ sender: UITapGestureRecognizer) {
+        let storyboard = UIStoryboard(name: "EmailAuthViewController", bundle: nil)
+        guard let vc = storyboard.instantiateViewController(withIdentifier: "e-mail auth"  ) as? EmailAuthViewController else {
+            return
+        }
         disableNotifications()
         self.present(vc, animated: true, completion: nil)
     }
     
-    @objc func googleButtonTap(_ sender: UITapGestureRecognizer) {
+    @objc
+    private func googleButtonTap(_ sender: UITapGestureRecognizer) {
         startGoogleLoader()
-        signInGoogleView.sendActions(for: .touchUpInside)
+        googleSignInView.sendActions(for: .touchUpInside)
+    }
+
+    private func enableNotifications() {
+        NotificationCenter.default.addObserver(self, selector: #selector(googleUserDidSignIn(_:)), name: .googleSignedIn, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(googleErrorAuth(_:)), name: .googleError, object: nil)
     }
     
-    func startGoogleLoader() {
-        UIView.animate(withDuration: Constants.forAnimation.normal) {
-            self.spinnerView.alpha = Constants.magicNumbers.fullAlpha
-            self.disableButtons()
-        }
+    private func disableNotifications() {
+        NotificationCenter.default.removeObserver(self, name: .googleSignedIn, object: nil)
+        NotificationCenter.default.removeObserver(self, name: .googleError, object: nil)
     }
     
-    func stopGoogleLoader() {
+    private func startGoogleLoader() {
         UIView.animate(withDuration: Constants.forAnimation.normal) {
-            self.spinnerView.alpha = Constants.magicNumbers.zeroAlpha
-            self.enableButtons()
+            self.spinner.startAnimating()
+            self.disableButtons()
         }
     }
     
-    func isUserLogged() -> Bool {
-        guard let _ = GIDSignIn.sharedInstance()?.currentUser else {
-            return false
+    private func stopGoogleLoader() {
+        UIView.animate(withDuration: Constants.forAnimation.fast) {
+            self.spinner.stopAnimating()
+            self.enableButtons()
         }
-        return true
     }
-    
-    func enableButtons() {
+
+    private func enableButtons() {
         googleAuth.isUserInteractionEnabled = true
-        googleAuth.alpha = Constants.magicNumbers.fullAlpha
+        googleAuth.alpha = 1
         emailAuth.isUserInteractionEnabled = true
-        emailAuth.alpha = Constants.magicNumbers.fullAlpha
+        emailAuth.alpha = 1
     }
     
-    func disableButtons() {
+    private func disableButtons() {
         googleAuth.isUserInteractionEnabled = false
-        googleAuth.alpha = Constants.magicNumbers.halfAlpha
+        googleAuth.alpha = 0
         emailAuth.isUserInteractionEnabled = false
-        emailAuth.alpha = Constants.magicNumbers.halfAlpha
+        emailAuth.alpha = 0.5
```

Comment:
0.5 still a magic number
make constant in this class with name "disableTransparencyAlpha"